### PR TITLE
Updated all examples to be constructor injection rather than field injection.

### LINKED ISSUE #1068: Change examples and tutorials to constructor injection
per spring engineer: "field injection is evil" http://olivergierke.de/2013/11/why-field-injection-is-evil/

### PR DESCRIPTION
**Pull request by dogeared**
Fixes #1068

### CODE REVIEW

File: examples/spring-boot-default/src/main/java/com/stormpath/spring/boot/examples/HelloController.java [hunk 1]
```diff
@@ -29,9 +29,13 @@
 @Controller
 public class HelloController {
 
-    @Autowired
     private HelloService helloService;
```
Comment (mrioan) [inline examples/spring-boot-default/src/main/java/com/stormpath/spring/boot/examples/HelloController.java]: Can't we just keep the `@Autowired` field and add a `notNull` assertion  in a constructor with NO arguments?. The outcome will be the same with less code to read, and less impact to the interface (as a new contractor with parameters requires changes in the code that creates it).

File: examples/spring-boot-default/src/main/java/com/stormpath/spring/boot/examples/HelloController.java [hunk 2]
```diff
@@ -29,9 +29,13 @@
...
 
+    @Autowired
+    public HelloController(HelloService helloService) {
+        this.helloService = helloService;
```
Comment (mrioan) [inline examples/spring-boot-default/src/main/java/com/stormpath/spring/boot/examples/HelloController.java]: This is missing a `notNull` assertion

File: examples/spring-security-spring-boot-webmvc/src/main/java/com/stormpath/spring/boot/examples/HelloController.java [hunk 3]
```diff
@@ -30,9 +30,13 @@
 @Controller
 public class HelloController {
 
-    @Autowired
     private HelloService helloService;
 
+    @Autowired
+    public HelloController(HelloService helloService) {
+        this.helloService = helloService;
```
Comment (mrioan) [inline examples/spring-security-spring-boot-webmvc/src/main/java/com/stormpath/spring/boot/examples/HelloController.java]: Missing a `notNull` assertion. There are more like this below. I will not flag them all so please take a look at the other ones yourself. Thanks !

File: examples/spring-boot-default/src/main/java/com/stormpath/spring/boot/examples/HelloController.java (hunk 1 above)
Comment (dogeared) [inline examples/spring-boot-default/src/main/java/com/stormpath/spring/boot/examples/HelloController.java]: According to the article, having an explicit constructor improves testability as you can instantiate the object and pass in a mock as the injected dependency.

File: examples/spring-boot-default/src/main/java/com/stormpath/spring/boot/examples/HelloController.java (hunk 2 above)
Comment (dogeared) [inline examples/spring-boot-default/src/main/java/com/stormpath/spring/boot/examples/HelloController.java]: It's not necessarily required to have the not null check. Do you think it's important that they all have the check?

File: examples/spring-boot-default/src/main/java/com/stormpath/spring/boot/examples/HelloController.java (hunk 1 above)
Comment (mrioan) [inline examples/spring-boot-default/src/main/java/com/stormpath/spring/boot/examples/HelloController.java]: I kinda agree with you BUUUUT we are simply providing examples for People to use our product. From my point of view the examples need to be as readable as possible. A  working example that is simple to read is heaven to me. I do not think we should we decreasing readability in favor of following a good practice that we do not really need as those examples are not even Unit Tested (therefore the constructor is never really customized by tests). They are just run by ITs. To be honest I do not think we should be changing tests to follow this practices, it adds complexity and provides no real gain.

**Comment by dogeared** [comment]
@mrioan - per conversation, only the tutorial section has been changed to use constructor injection

**Comment by mrioan** [comment]
@dogeared approved. Back to you. Thanks!

**Pull request merged**